src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -622,71 +629,147 @@ private AIState waitForShovel()
+     * @param tool tool required for block
+     * @return true if we need a tool
+     */
+    private boolean checkForNeededTool(@NotNull final IToolType toolType, final int minimalLevel)
+    {
+        final int maxToolLevel = worker.getWorkBuilding().getMaxToolLevel();
+        final InventoryCitizen inventory = worker.getInventoryCitizen();
+        if (InventoryUtils.isToolInItemHandler(new InvWrapper(inventory), toolType, minimalLevel, maxToolLevel))
         {
             return false;
         }
+
         delay += DELAY_RECHECK;
         if (walkToBuilding())
         {
             return true;
         }
-        if (isToolInHut(tool))
+        if (isToolInHut(toolType, minimalLevel))
         {
             return false;
         }
         if (!getOwnBuilding().hasOnGoingDelivery())
         {
-            chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_TOOLREQUEST, tool, InventoryUtils.swapToolGrade(hutLevel));
+            chatRequestTool(toolType, minimalLevel, maxToolLevel);
+
         }
         return true;
     }
 
+    private void chatRequestTool(@NotNull final IToolType toolType, final int minimalLevel, final int maximumLevel)
+    {
+        if (minimalLevel == TOOL_LEVEL_WOOD_OR_GOLD)
+        {
+            if (maximumLevel == TOOL_LEVEL_MAXIMUM)
+            {
+                //Any tool level will do
+                chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_SIMPLETOOLREQUEST, toolType.getName());
+            }
+            else
+            {
+                // tools at most ...
+                if (toolType.hasMaterial())
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_TOOLREQUEST, toolType.getName(), ItemStackUtils.swapToolGrade(maximumLevel));
+                }
+                else
+                {
+                    chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_ENCHTOOLREQUEST, toolType.getName(), maximumLevel-1);
+                }
+            }
+        }
+        else if (maximumLevel == TOOL_LEVEL_MAXIMUM)
+        {
+            // at least
+            chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_TOOLATLEASTREQUEST, toolType.getName(), minimalLevel);
+
+        }
+        else if (minimalLevel > maximumLevel)
+        {
+            // need to upgrade the worker's hut
+            chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_PICKAXEREQUESTBETTERHUT, minimalLevel - 1);
+        }
+        else if (minimalLevel == maximumLevel)
+        {
+            // we need a specific grade
+            chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_SPECIFICTOOLREQUEST, toolType.getName());
+        }
+        else
+        {
+           // at least and at most
+           chatSpamFilter.talkWithoutSpam(COM_MINECOLONIES_COREMOD_ENTITY_WORKER_PICKAXEREQUEST,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this line start at column 13. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)